src/algorithm/REDUCE_SUM-Cuda.cpp

```diff
@@ -141,31 +139,38 @@ void REDUCE_SUM::runCudaVariantBlock(VariantID vid)
 
   if ( vid == Base_CUDA ) {
 
-    Real_ptr dsum;
-    allocData(DataSpace::CudaDevice, dsum, 1);
+    RAJAPERF_GPU_REDUCER_SETUP(Real_ptr, sum, hsum, 1);
 
     startTimer();
     for (RepIndex_type irep = 0; irep < run_reps; ++irep) {
 
-      cudaErrchk( cudaMemcpyAsync( dsum, &m_sum_init, sizeof(Real_type),
-                                   cudaMemcpyHostToDevice, res.get_stream() ) );
+      if (sum != hsum) {
```

Comment:
Can this part and the copy back after the kernel also be made into a general macro if there are adequate args passed to distinguish the various cases?

Reply:
Ok, I did this.